### ISSUE TITLE
simplewallet infity loop on Ctrl-D

### ISSUE THREAD

**Issue by fanatid** [monero-project/bitmonero#771]
```
./simplewallet
..
^D
Specify wallet file name (e.g., MyWallet). If the wallet doesn't exist, it will be created.
Wallet file name: Error: wallet file path not valid: 
Specify wallet file name (e.g., MyWallet). If the wallet doesn't exist, it will be created.
Wallet file name: Error: wallet file path not valid: 
```
it generate infinity loop, because [ask_wallet_create_if_needed uses do-while without exit condition](https://github.com/monero-project/bitmonero/blob/0ee87e63050c31405182042868a89d4f3f3ac7c2/src/simplewallet/simplewallet.cpp#L696)

**Comment by moneromooo-monero** [comment]
https://github.com/monero-project/bitmonero/pull/772

**Comment by fluffypony** [comment]
Fixed via #772
